### PR TITLE
Fix radar rendering for heroes and castles

### PR DESCRIPTION
**Pull request by idshibanov**
Relates to #6845 .

Castles and heroes must be synced up with `vec_heroes` and `vec_castles` world objects to be properly rendered on the map.

### CODE REVIEW

File: src/fheroes2/editor/editor_interface.cpp [hunk 1]
```diff
@@ -1062,6 +1063,12 @@ namespace Interface
 
             setObjectOnTileAsAction( tile, groupType, _editorPanel.getSelectedObjectType() );
 
+            Heroes * hero = world.GetHeroForHire( Race::IndexToRace( static_cast<int>( objectInfo.metadata[1] ) ) );
+            if ( hero ) {
+                hero->SetCenter( tilePos );
+                hero->SetColor( static_cast<int>( color ) );
```
Comment (Districh-ru) [inline src/fheroes2/editor/editor_interface.cpp:1069]: We should convert color index here:
```suggestion
                hero->SetColor( static_cast<int>( Color::IndexToColor( static_cast<int>( color ) ) ) );
```

**Review by Districh-ru — COMMENTED** [review]
Hi, @idshibanov, I left one comment, could you please take a look when you have time?

Possibly in the future we may improve radar "for Editor" to use `Map_Format::MapFormat & map` instead of `world` to get objects' data for render.

**Review by Districh-ru — APPROVED** [review]

File: src/fheroes2/maps/map_format_helper.cpp [hunk 2]
```diff
@@ -125,12 +126,32 @@ namespace Maps
...
+                else if ( object.group == Maps::ObjectGroup::KINGDOM_HEROES ) {
+                    const auto & metadata = heroObjects[object.index].metadata;
+                    const uint8_t color = Color::IndexToColor( static_cast<int>( metadata[0] ) );
+
+                    Heroes * hero = world.GetHeroForHire( static_cast<int>( metadata[1] ) );
+                    if ( hero ) {
+                        hero->SetCenter( { static_cast<int32_t>( i ) % world.w(), static_cast<int32_t>( i ) / world.w() } );
+                        hero->SetColor( color );
+                    }
+                }
```
Comment (ihhub) [inline src/fheroes2/maps/map_format_helper.cpp:154]: This change breaks map loading for gameplay. Such logic shouldn't exist here as this logic is within world_loadmap.cpp file.

Comment (ihhub) [reply src/fheroes2/maps/map_format_helper.cpp:154]: `map_format_helper.cpp` file should not contain any logic related to World itself. It should only help with loading information from the map format. The rest of logic is set in `world_loadmap.cpp`. We can move the proposed logic for the Editor only.

Comment (idshibanov) [reply src/fheroes2/maps/map_format_helper.cpp:154]: @ihhub Is there a place for MapFormat to be stored during gameplay? Not within editor files or loadmap. I don't see a way for Radar to access `MapFormat & map` currently.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@idshibanov , thank you so much for the fix!

**Pull request merged**